check-repo-links

```diff
@@ -2,14 +2,11 @@
 # Check repo links
 
 DISPOSITION=0
-for charm in $(cat charms.txt); do
+charms=$(cd charms && ls -d1 *)
+for charm in $charms; do
 
-  # Handle repo url overrides if present
-  if grep "^$charm|" repo-link-overrides.txt > /dev/null; then
-     REPO_URL=$(grep "^$charm|" repo-link-overrides.txt | cut -f 2 -d "|")
-  else
-     REPO_URL="https://opendev.org/openstack/charm-${charm}"
-  fi
+  TLP=$(grep "^project=.*" charms/$charm/.gitreview | perl -n -e'/=(.*)\// && print $1')
+  REPO_URL="https://opendev.org/${TLP}/charm-${charm}"
```

Comment:
we should re-use what ever is in the .gitreview file instead of assuming the format `charm-${charm}`.

This is a way we could grab the piece that goes after `opendev.org`:

```
$ REPO=$(grep "^project=.*" .gitreview | awk -F'=' '{print $2}')
$ echo $REPO
openstack/charm-octavia.git
$ echo ${REPO%.*}
openstack/charm-octavia
```

Reply:
Bug raised #227